k8sutils/statefulset.go

```diff
@@ -171,12 +176,17 @@ func createPVCTemplate(stsMeta metav1.ObjectMeta, storageSpec corev1.PersistentV
 	} else {
 		pvcTemplate.Spec.AccessModes = storageSpec.Spec.AccessModes
 	}
+	pvcVolumeMode := corev1.PersistentVolumeFilesystem
+	if storageSpec.Spec.VolumeMode != nil {
+		pvcVolumeMode = *storageSpec.Spec.VolumeMode
+	}
```

Comment:
If no VolumeMode is set, assume `FileSystem`... again, not ideal... but if one is not set, we get a constant update.